lib/LTP/WhiteList.pm

```diff
@@ -33,14 +37,17 @@ sub download_whitelist {
     my $res = Mojo::UserAgent->new->get($path)->result;
     unless ($res->is_success) {
         record_info("File not downloaded!", $res->message, result => 'softfail');
-        set_var('LTP_KNOWN_ISSUES', undef);
+        set_var('LTP_KNOWN_ISSUES_LOCAL', undef);
         return;
     }
+
     my $basename = $path =~ s#.*/([^/]+)#$1#r;
-    save_tmp_file($basename, $res->body);
-    set_var('LTP_KNOWN_ISSUES', hashed_string($basename));
+
     mkdir('ulogs') if (!-d 'ulogs');
-    bmwqemu::save_json_file($res->json, "ulogs/$basename");
+    save_tmp_file($basename, $res->body);
+    copy(hashed_string($basename), "ulogs/$basename");
+
+    set_var('LTP_KNOWN_ISSUES_LOCAL', "ulogs/$basename");
```

Comment:
Reading whitelist data from the `ulogs` copy makes the `save_tmp_file()` unnecessary. But I believe that the temp file was created because the `ulogs` copy might get accidentally overwritten if one of the test modules uploads a log file with the same filename.

Reply:
OK, I'll just create file directly:

```
mkdir('ulogs') if (!-d 'ulogs');
my $file = "ulogs/$basename";
open(my $fh, '>', $file) or die "Could not open file '$file' $!";
print $fh $res->body;
close $fh;

```

Reply:
I'm talking about setting `LTP_KNOWN_ISSUES_LOCAL` to `"ulogs/$basename"` instead of `hashed_string($basename)`. The file writes are fine.